File: Python/cc12.py
"""
Problem: Replace Digits with Asterisks
Difficulty: Easy

Given a string s, replace every digit character (0 through 9) with an asterisk (*)
and return the modified string.

Examples:
----------
Input: s = "abc123"
Output: "abc***"

Input: s = "passw0rd123!"
Output: "passw*rd***!"

Constraints:
------------
•  1 <= len(s) <= 1000
•  s may contain letters, digits, punctuation, or symbols
"""

# define a function that takes one parameter s
#def hidesNumbers(s):
#   declare a list of numbers from 0 - 9 
#   numbers = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"]
#   declare modifiedWord string and set to empty ""
#   modifiedWord = ""
#   use a for loop to iterate through s string
#   for i in s:
#       use if statement to check for the condition
#       if i in numbers:
#           modifiedWord += "*"
#       else if no i was a match within numbers return add i to modifiedWord
#       else:
#            modifiedWord += i
#
#   return modifiedWord 
# 
# print(hidesNumbers(s))
          

def hidesNumbers(s):
    numbers = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"]
    modifiedWord = ""
    for i in s:
        if i in numbers:
            modifiedWord += "*"
        else:
            modifiedWord += i
    
    return modifiedWord

print(hidesNumbers("2aksjfkj1shdfkja3"))

File: JavaScript/cc12.js
// * LeetCode-style Problem: Square Each Number
//  *
//  * Given an array of integers `nums`, return a new array where each element
//  * is the square of the corresponding element in the input array.
//  *
//  * Parameters:
//  * int[] nums - a non-empty array of integers
//  *
//  * Returns:
//  * int[] - an array containing the squares of the input numbers
//  *
//  * Examples:
//  * Input: nums = [1, 2, 3, 4]
//  * Output: [1, 4, 9, 16]
//  *
//  * Input: nums = [-1, -2, -3]
//  * Output: [1, 4, 9]
//  *
//  * Input: nums = [0]
//  * Output: [0]
//  *
//  * Constraints:
//  * - 1 <= nums.length <= 10^4
//  * - -10^3 <= nums[i] <= 10^3
//  */

// create a function called squareNums passing 1 parameter nums
// function squareNums(nums) {
//      declare a variable a variable called squaredR let type and set to []   
//      let squaredR = [];
//      use a for loop to iterate on nums
//      for (let i = 0; i < nums.length; i++) {
//          use Math.pow( to raise the number times, pass (nums[i], 2). Use .push() to add all numbers into squaredR
//          squaredR.push(Math.pow(nums[i], 2));  
//      }
//      return squaredArray;
//}

function squareNums(nums) {
    let squaredR = [];
    for (let num of nums) {
        squaredR.push(Math.pow(num, 2));   
    }
    return squaredR;
}

console.log(squareNums([-1, -2, -3, 4]));